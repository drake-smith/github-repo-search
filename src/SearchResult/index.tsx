import React from 'react';
import { Link } from 'react-router-dom';

/*
 * For purpose of this project and time, allow any type here to define the API response object,
 * which contains more data than we need.
 */
interface SearchResultProps {
  id: number;
  name: string;
  login: string;
  description: string;
  stargazers_count: number;
  language: string;
}

const SearchResult = ({
  id,
  name,
  login,
  description,
  stargazers_count,
  language,
}: SearchResultProps) => {
  return (
    <li key={id}>
      <Link to={`${id}`}>
        <h2>{name}</h2>
      </Link>
      <p>Owner: {login}</p>
      <p>{description}</p>
      <p>Number of stars: {stargazers_count}</p>
      <p>Language: {language}</p>
    </li>
  );
};

export default SearchResult;
