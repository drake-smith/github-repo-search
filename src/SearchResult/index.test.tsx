import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import SearchResult from '.';

describe('<SearchResult />', () => {
  const handleSelectSearchResult = jest.fn();

  test('It renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SearchResult
            key={24736}
            index={3}
            handleSelectSearchResult={handleSelectSearchResult}
            id={24736}
            name={'Repository'}
            login={'John Smith'}
            description={'This is a description'}
            stargazers_count={500}
            language={'JavaScript'}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('handleSelectSearchResult is called correctly', () => {
    render(
      <MemoryRouter>
        <SearchResult
          key={24736}
          index={3}
          handleSelectSearchResult={handleSelectSearchResult}
          id={24736}
          name={'Repository'}
          login={'John Smith'}
          description={'This is a description'}
          stargazers_count={500}
          language={'JavaScript'}
        />
      </MemoryRouter>
    );
    const linkElem = screen.getByText('Repository');
    fireEvent.click(linkElem);
    expect(handleSelectSearchResult).toHaveBeenCalledTimes(1);
  });
});
