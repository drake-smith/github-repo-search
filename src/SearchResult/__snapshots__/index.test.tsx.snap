// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> It renders correctly 1`] = `
<li
  className="sc-bdvvtL bBGwRo"
>
  <div>
    <a
      className="sc-gsDKAQ peeim"
      href="/24736"
      onClick={[Function]}
    >
      <h2
        className="sc-dkPtRN fCvFVC"
      >
        Repository
      </h2>
    </a>
    <p
      className="sc-hKwDye jzfCYy"
    >
      This is a description
    </p>
  </div>
  <div>
    <div
      className="sc-eCImPb iMXmGk"
    >
      <p>
        by 
        <strong>
          John Smith
        </strong>
      </p>
      <p>
        500
         ‚≠ê 
      </p>
      <p>
        JavaScript
      </p>
    </div>
  </div>
</li>
`;
